examples/step-62/step-62.cc

```diff
@@ -1055,6 +1057,13 @@ namespace step62
     // displacement_data contains the value of the displacement at these points.
     std::vector<hsize_t>              coordinates;
     std::vector<std::complex<double>> displacement_data;
+
+    auto                       mapping = StaticMappingQ1<dim>::mapping;
```

Comment:
If you rebase, this should now be as follows:
```suggestion
    auto                       mapping = get_default_linear_mapping(triangulation);
```

Reply:
I added `get_default_linear_mapping` as a _todo comment_. I think that the branch of luca that I use as base is too old. I'll update this once the branch of luca is in master

Reply:
@bangerth I rebased and as you suggested I tried to use the following line of code:
```c++
auto                       mapping = get_default_linear_mapping(triangulation);
```
But it I use that line of code I get the following. Do you know why I get this compilation error?
```cmake
examples/step-62/step-62.cc:1061:32: error: variable type 'dealii::Mapping<2, 2>' is an abstract class                                                        
    auto                       mapping = get_default_linear_mapping(triangulation);                                                                                                            
                               ^                                                                                                                                                               
examples/step-62/step-62.cc:1309:9: note: in instantiation of member function 'step62::ElasticWave<2>::store_frequency_step_data' requested here              
        store_frequency_step_data(frequency_idx);                                                                                                                                              
        ^                                                                                                                                                                                      
examples/step-62/step-62.cc:1365:5: note: in instantiation of member function 'step62::ElasticWave<2>::frequency_sweep' requested here                        
    frequency_sweep();                                                                                                                                                                         
    ^                                                                                                                                                                                          
examples/step-62/step-62.cc:1487:25: note: in instantiation of member function 'step62::ElasticWave<2>::run' requested here                                   
        elastic_problem.run();                                                                                                                                                                 
                        ^                                                                                                                                                                      
include/deal.II/fe/mapping.h:321:3: note: unimplemented pure virtual method 'clone' in 'Mapping'                                                              
  clone() const = 0;                                                                                                                                                                           
  ^                                                                                                                                                                                            
include/deal.II/fe/mapping.h:402:3: note: unimplemented pure virtual method 'preserves_vertex_locations' in 'Mapping'                                         
  preserves_vertex_locations() const = 0;                                                                                                                                                      
  ^                                                                                                                                                                                            
include/deal.II/fe/mapping.h:409:3: note: unimplemented pure virtual method 'is_compatible_with' in 'Mapping'                                                 
  is_compatible_with(const ReferenceCell &reference_cell) const = 0;                                                                                                                           
  ^                                                                                                                                                                                            
include/deal.II/fe/mapping.h:428:3: note: unimplemented pure virtual method 'transform_unit_to_real_cell' in 'Mapping'                                        
  transform_unit_to_real_cell(                                                                                                                                                                 
  ^                                                                                                                                                                                            
include/deal.II/fe/mapping.h:461:3: note: unimplemented pure virtual method 'transform_real_to_unit_cell' in 'Mapping'                                        
  transform_real_to_unit_cell(                                                                                                                                                                 
  ^                                                                                                                                                                                            
include/deal.II/fe/mapping.h:687:3: note: unimplemented pure virtual method 'requires_update_flags' in 'Mapping'                                              
  requires_update_flags(const UpdateFlags update_flags) const = 0;                                                                                                                             
  ^                                                                                                                                                                                            
include/deal.II/fe/mapping.h:738:3: note: unimplemented pure virtual method 'get_data' in 'Mapping'                                                           
  get_data(const UpdateFlags      update_flags,                                                                                                                                                
  ^                                                                                                                                                                                            
include/deal.II/fe/mapping.h:808:3: note: unimplemented pure virtual method 'get_subface_data' in 'Mapping'                                                   
  get_subface_data(const UpdateFlags          update_flags,                                                                                                                                    
  ^                                                                                                                                                                                            
include/deal.II/fe/mapping.h:895:3: note: unimplemented pure virtual method 'fill_fe_values' in 'Mapping'                                                     
  fill_fe_values(                                                                                                                                                                              
  ^
include/deal.II/fe/mapping.h:975:3: note: unimplemented pure virtual method 'fill_fe_subface_values' in 'Mapping'
  fill_fe_subface_values(
  ^
include/deal.II/fe/mapping.h:1058:3: note: unimplemented pure virtual method 'transform' in 'Mapping'
  transform(const ArrayView<const Tensor<1, dim>> &                  input,
  ^
include/deal.II/fe/mapping.h:1111:3: note: unimplemented pure virtual method 'transform' in 'Mapping'
  transform(const ArrayView<const DerivativeForm<1, dim, spacedim>> &input,
  ^
include/deal.II/fe/mapping.h:1169:3: note: unimplemented pure virtual method 'transform' in 'Mapping'
  transform(const ArrayView<const Tensor<2, dim>> &                  input,
  ^
include/deal.II/fe/mapping.h:1216:3: note: unimplemented pure virtual method 'transform' in 'Mapping'
  transform(const ArrayView<const DerivativeForm<2, dim, spacedim>> &input,
  ^
include/deal.II/fe/mapping.h:1269:3: note: unimplemented pure virtual method 'transform' in 'Mapping'
  transform(const ArrayView<const Tensor<3, dim>> &                  input,
  ^
1 error generated.
```

Reply:
You probably need to write this as
```
const auto &mapping = get_default_linear_mapping(triangulation);
```
What you had is probably trying to store the mapping in an instance of the abstract class `Mapping<dim, spacedim>`, which is not permitted. The `get_default_linear_mapping()` function returns a reference to a concrete instance of a mapping.

Reply:
Thanks a lot @jppelteret, that was a really quick reply!
Now it works :)